Initial ACDD Tests

abkfenris opened the pull request:
I wanted to share an early draft PR before implementing too many rules to make sure I'm on the right track. 

The IOOS compliance checker has implemented their own tests for ACDD, CF, IOOS, and others, but have ended up with a much more complex and hard to adapt library in the process. There is interest in using XRLint to help simplify things, and make it easier to share work across the ecosystem. Along those lines an idea is to move tests/rules upstream if there is interest rather than keeping their own implementation.

I've converted many of the more general ACDD global attribute tests over here.

abkfenris committed:
Initial ACDD Tests

An initial set of ACDD tests.

Adapted from ioos/compliance_checker.

forman commented:
@abkfenris, thank you very much and we apologize for overlooking this. We are extremely busy at the moment, which is why `xrlint`'s maintenance has been somewhat neglected.

abkfenris commented:
@forman Glad to hear that `xrlint` is still something you're working on. 

We're still interested in implementing tests for ACDD and other conventions, but didn't want to go too far down the wrong track or sink a lot of effort into something that is no longer maintained.

forman commented:
It is in active use in our company (for sure a few others), just maintenance slowed down recently. We'll be back!

abkfenris commented:
Great! When I went asking around the community many folks hadn't heard about it but were very interested.

forman commented on pyproject.toml:
```diff
@@ -64,6 +64,11 @@ exclude = ["**/*.ipynb"]
 
 [project.optional-dependencies]
+acdd = [
+    "isodate>=0.7.2",
```
I'm also fine to make this a common project dependency.

forman reviewed (commented):
That looks already very good. Great work! We are happy to take this on board.

To move forward, could you please also
- Describe in `CHANGES.md` what will we added
- Adapt the rule documentation generator [mkruleref.py](https://github.com/bcdev/xrlint/blob/main/mkruleref.py) to also convert the new rules into markdown docs.
- Run the converter to update [docs/rule-ref.md](https://github.com/bcdev/xrlint/blob/main/docs/rule-ref.md)

Note, I wont be able to support you until the 14.01.2026, therefore pls contact my colleague @clarasb.

abkfenris commented:
Great, I'll work on getting those ready by the time you're back, and I may have some additional ACDD tests adapted as well.

forman commented:
...and please add yourself to the list of authors in `pyproject.toml`.

abkfenris committed:
Clean up initial ACDD rules

Adds documentation for ACDD tests, isodate to the dependencies, and updates the changes.

clarasb commented on tests/plugins/acdd/rules/test_attributes.py:
```diff
@@ -0,0 +1,34 @@
+
+valid_1_3_highly_rec_dataset = xr.Dataset(
+    attrs={
+        "title": "Tis only a test",
```
```suggestion
        "title": "This is only a test",
```

abkfenris committed:
Switch pyproject back to optional-dependencies

Looks like pip in tests isn’t ready for the newer `dependency-groups` for development tooling that uv prefers.

clarasb commented on tests/plugins/acdd/rules/test_attributes.py:
```diff
@@ -0,0 +1,34 @@
+
+valid_1_3_highly_rec_dataset = xr.Dataset(
+    attrs={
+        "title": "Tis only a test",
```
```suggestion
        "title": "This is only a test",
```

abkfenris replied:
Thanks! This is why my team jokes that we could build our own version of `git blame` but based around all of our team members usual ways of misspelling things.

clarasb reviewed (commented):
Thanks @abkfenris ! That looks good to me. 

Please consider
- the change suggestion, 
- to fix the `reccomended` typo,
- and to add the `acdd` plugin to the Readme.

Then, let's wait for a final review from @forman.

abkfenris committed:
Update tests/plugins/acdd/rules/test_attributes.py

Co-authored-by: Clara Backens <64074334+clarasb@users.noreply.github.com>

abkfenris committed:
Fix reccomended misspellings

forman reviewed (approved):
¡Oki!

forman committed:
Merge branch 'main' into acdd
